book/working-with-multiple-records.md

```diff
@@ -0,0 +1,201 @@
+Working With Multiple Records
+=============================
+
+Sometimes one request can create or affect several records in one or more tables
+in such cases its advisable to follow [ACID](https://en.wikipedia.org/wiki/ACID)
+rules. Yii2 supports transactions, isolation levels, and allow you to
+validate data independently, so following the ACID rules is simple.
+
+For example when creating a credit, you also need to store the credit references
+and file associated to it.
+
+```php
+public function actionCreate()
+{
+    $model = new Credit();
+    $modelReferences = [new CreditRefence()];
+    $modelFiles = [new CreditFile()];
+
+    if (Yii::$app->request->isPost) {
+        $model->load(Yii::$app->request->post());
+
+        foreach (Yii::$app->requestPost($modelReferences[0]->formName())
+            as $i => $data
+        ) {
+            $newModel = new CreditReference();
+            $newModel->load($data, '');
+            $modelReferences[$i] = $newModel;
+        }
+
+        foreach (Yii::$app->requestPost($modelFiles[0]->formName())
+            as $i => $data
+        ) {
+            $newModel = new CreditFile();
+            $newModel->load($data, '');
+            $newModel->file = UploadedFile::getInstance($newModel, "[$i]file");
+            $modelFiles[$i] = $newModel;
+        }
+
+        $transaction = Yii::$app->db->beginTransaction(
```

Comment:
before loading the data to the models? It can be done for the cookbook for sure, i personally don't want the transaction to stay open for too long since it might cause other requests using the same tables to wait till this transaction is completed.